bcolz/carray_ext.pyx

```diff
@@ -1216,9 +1229,17 @@ cdef class carray:
             self.chunks.append(chunk_)
             cbytes += chunk_.cbytes
         self.leftover = leftover = nbytes % self._chunksize
+
```

Comment:
I remembered why I added this, leftovers were not iterated for in memory carray, I updated the tests, but this is for sure what is causing the problems.
With this update on the tests https://github.com/FrancescElies/bcolz/commit/541e3abee475f94f13495b1481bc5b431b5d6d3c if you remove that code and rerun the suite you will see that in memory carray does not iterate over the leftovers.
```
FAIL: test00 (test_carray.chunkIterMemoryTest)
Testing chunk iterator including leftovers
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/francescelieshenar/01_working_dir/01_github/bcolz/bcolz/tests/test_carray.py", line 2164, in test00
    self.assertEquals(n, len(a) / chunklen_)
AssertionError: 9 != 10
```
PS.: you will see another failure, I don't know exactly why, but this is for sure the code causing problems